lib/ClangImporter/ClangAdapter.cpp

```diff
@@ -702,19 +702,28 @@ bool importer::isUnavailableInSwift(
   return false;
 }
 
-OptionalTypeKind importer::getParamOptionality(const clang::ParmVarDecl *param,
+OptionalTypeKind importer::getParamOptionality(version::Version swiftVersion,
+                                               const clang::ParmVarDecl *param,
                                                bool knownNonNull) {
   auto &clangCtx = param->getASTContext();
 
   // If nullability is available on the type, use it.
-  if (auto nullability = param->getType()->getNullability(clangCtx)) {
+  clang::QualType paramTy = param->getType();
+  if (auto nullability = paramTy->getNullability(clangCtx)) {
     return translateNullability(*nullability);
   }
 
   // If it's known non-null, use that.
   if (knownNonNull || param->hasAttr<clang::NonNullAttr>())
     return OTK_None;
 
+  // Check for the 'static' annotation on C arrays.
+  if (!swiftVersion.isVersion3())
+    if (const auto *DT = dyn_cast<clang::DecayedType>(paramTy))
```

Comment:
I guess in the C world you don't say "C > 99" or something. It's only because we have sequential version numbers that we even have this option.

In Clang we check for the new version of the language (`if (LangOpts.C99)`, `if (LangOpts.CXX14)`), but those checks are designed to stay around. Checking `isVersion3` feels like a reminder that the check will go away if/when we drop support for Swift 3 mode.